nixops/deployment.py

```diff
@@ -440,16 +439,15 @@ def evaluate_args(self) -> Any:
         except subprocess.CalledProcessError:
             raise NixEvalError
 
-    def evaluate_config(self, attr):
+    def evaluate_config(self, attr) -> Dict:
         try:
-            # FIXME: use --json
-            xml = subprocess.check_output(
+            _json = subprocess.check_output(
                 ["nix-instantiate"]
                 + self.extra_nix_eval_flags
                 + self._eval_flags(self.nix_exprs)
                 + [
                     "--eval-only",
-                    "--xml",
+                    "--json",
```

Comment:
Unfortunately this is not quite equivalent:

```
[grahamc@Petunia:~]$ nix-instantiate  --readonly-mode --eval-only --xml --strict ./test.nix
<?xml version='1.0' encoding='utf-8'?>
<expr>
  <attrs>
    <attr column="3" line="2" name="foo" path="/home/grahamc/test.nix">
      <path value="/home/grahamc/test.nix" />
    </attr>
  </attrs>
</expr>

[grahamc@Petunia:~]$ nix-instantiate  --readonly-mode --eval-only --json --strict ./test.nix
{"foo":"/nix/store/ml1vsmamxm0ijbc67g7za455n3g75j1l-test.nix"}%                                                                                                                                                     
```

I wonder if there is a way to get the XML behavior under the JSON output mode.

Reply:
or changing the `keyFile`'s module behavior to `apply` `tostring` to the file paths